### PR TITLE
Add support for integration testing runtime builder image

### PR DESCRIPTION
**Pull request by nkubala**
This adds support for integration testing the runtime builder images, as part of the gcloud app deploy flow. The sample application will be provided with a templated builder `test.yaml.in` file (as opposed to a templated `Dockerfile.in`), which will have the staging builder image inserted into it, and used as the runtime builder via setting a gcloud config var.

Some runtimes (such as golang) do not even have a base runtime image to integration test, so this is a necessary feature for e2e testing the silver languages.

@dlorenc

### CODE REVIEW

File: integration_tests/testsuite/deploy_app.py [hunk 1]
```diff
@@ -38,21 +38,39 @@ def _set_app_image(image):
     fin.close()
 
 
-def deploy_app(image, appdir):
+def _set_builder_image(builder):
+    with open('test.yaml.in', 'r') as fin:
+        with open('test.yaml', 'w') as fout:
+            for line in fin:
+                fout.write(line.replace('${STAGING_BUILDER_IMAGE}', builder))
+        fout.close()
```
Comment (dlorenc) [inline integration_tests/testsuite/deploy_app.py]: You don't need these close statements with the with statement.

File: integration_tests/testsuite/deploy_app.py [hunk 2]
```diff
@@ -38,21 +38,39 @@ def _set_app_image(image):
...
 
         deployed_version = test_util.generate_version()
 
         # TODO: once sdk driver is published, use it here
-        deploy_command = ['gcloud', 'app', 'deploy', '--no-promote'
+        deploy_command = ['gcloud', 'app', 'deploy', '--no-promote',
                           '--version', deployed_version, '-q']
+        if yaml:
+            logging.info(yaml)
+            deploy_command.append(yaml)
```
Comment (dlorenc) [inline integration_tests/testsuite/deploy_app.py]: Do you need a flag here, or is this just a positional?

Comment (nkubala) [inline integration_tests/testsuite/deploy_app.py]: positional, it's the last argument. if it isn't provided, gcloud assumes it's `<current_directory>/app.yaml`.

File: integration_tests/testsuite/deploy_app.py (hunk 1 above)
Comment (nkubala) [inline integration_tests/testsuite/deploy_app.py]: fixed

File: integration_tests/testsuite/deploy_app.py [hunk 3]
```diff
@@ -28,31 +28,45 @@ def _cleanup(appdir):
...
     with open('Dockerfile.in', 'r') as fin:
         with open('Dockerfile', 'w') as fout:
             for line in fin:
                 fout.write(line.replace('${STAGING_IMAGE}', image))
-        fout.close()
-    fin.close()
 
 
-def deploy_app(image, appdir):
+def _set_builder_image(builder):
```
Comment (dlorenc) [inline integration_tests/testsuite/deploy_app.py]: Would be nice to refactor this and set_base_image a bit.

**Pull request merged**